From c745464a9e04d6464fb520756e093bf430ccda11 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Thu, 2 Apr 2015 15:36:14 +0800
Subject: [PATCH] MLK-10555 ASoC: imx-wm8958: change AIF1CLK to 384fs to get
 accurate sample rate for 24 bit word length

change AIF1CLK to 384fs to get accurate sample rate for 24 bit word length.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
---
 sound/soc/fsl/imx-wm8958.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-wm8958.c b/sound/soc/fsl/imx-wm8958.c
index fb0399c..49ce823 100644
--- a/sound/soc/fsl/imx-wm8958.c
+++ b/sound/soc/fsl/imx-wm8958.c
@@ -163,7 +163,10 @@ static int imx_hifi_hw_params(struct snd_pcm_substream *substream,
 	} else {
 		data->sr_stream[tx] = params_rate(params);
 
-		pll_out = data->sr_stream[tx] * 256;
+		if (params_width(params) == 24)
+			pll_out = data->sr_stream[tx] * 384;
+		else
+			pll_out = data->sr_stream[tx] * 256;
 
 		ret = snd_soc_dai_set_pll(codec_dai, WM8994_FLL1,
 					  WM8994_FLL_SRC_MCLK1,
-- 
1.8.0

